object_detection/Mask_RCNN/mask_rcnn_run_loop.py

```diff
@@ -0,0 +1,244 @@
+flags.DEFINE_integer('ps_tasks', 0,
+                     'Number of parameter server tasks. If None, does not use '
+                     'a parameter server.')
+flags.DEFINE_string('train_dir', '',
+                    'Directory to save the checkpoints and training summaries.')
+
+flags.DEFINE_string('pipeline_config_path', '',
+                    'Path to a pipeline_pb2.TrainEvalPipelineConfig config '
+                    'file. If provided, other configs are ignored')
+
+flags.DEFINE_boolean('eval_training_data', False,
+                     'If training data should be evaluated for this job.')
+# check point dir is the same as the train dir
+# flags.DEFINE_string('checkpoint_dir', '',
+#                    'Directory containing checkpoints to evaluate, typically '
+#                    'set to `train_dir` used in the training job.')
+flags.DEFINE_string('eval_dir', '',
+                    'Directory to write eval summaries to.')
+
+flags.DEFINE_boolean('run_once', False, 'Option to only run a single pass of '
+                                        'evaluation. Overrides the `max_evals`'
+                                        ' parameter in the provided config.')
+flags.DEFINE_float('box_min_ap', -1, 'Option to run until the box average'
+                                    'precision reaches this number')
+flags.DEFINE_float('mask_min_ap', -1, 'Option to run until the mask average'
+                                     'precision reaches this number')
+FLAGS = flags.FLAGS
+
+
+def stopping_criteria_met(eval_metrics, mask_min_ap, box_min_ap):
+  """Returns true if both of the min precision criteria are met in the given
+  evaluation metrics.
+
+  Args:
+    eval_metrics: dict of metrics names as keys and their corresponding values,
+      containing "DetectionMasks_Precision/mAP", and
+      "DetectionBoxes_Precision/mAP" fields.
+    mask_min_ap: minimum desired mask average precision, will be ignored if -1
+    box_min_ap: minimum desired box average precision, will be ignored if -1
+
+  Returns:
+    True if non -1 criteria are met, false o.w.
+  """
+  assert mask_min_ap == -1 or 0 < mask_min_ap < 1
+  assert box_min_ap == -1 or 0 < box_min_ap < 1
+  try:
+    mask_mAP_reached = eval_metrics['DetectionMasks_Precision/mAP']
+    box_mAP_reached = eval_metrics['DetectionBoxes_Precision/mAP']
+  except KeyError as err:
+    raise Exception('eval_metrics dict does not contain the mAP field') from err
+
+  return (mask_min_ap == -1 or mask_mAP_reached > mask_min_ap) & \
+         (box_min_ap == -1 or box_mAP_reached > box_min_ap)
+
+
+def main(_):
+  assert FLAGS.train_dir, '`train_dir` is missing.'
+  assert FLAGS.pipeline_config_path, '`pipeline_config_path` is missing'
+  assert FLAGS.eval_dir, '`eval_dir` is missing.'
+
+  flags.DEFINE_string('checkpoint_dir', FLAGS.train_dir,
+                      'directory to read checkpoints from '
+                      'which is the same as the train_dir')
```

Comment:
Better not to define a flag here separately from all the other flags. If `checkpoint_dir` is almost always the same as `train_dir`, may as well just use `train_dir` instead.